### ISSUE TITLE
Planning the 0.6.4 release

### ISSUE THREAD

**Issue by Wohlstand** [libgme/game-music-emu#75]
I think, it's a time to release the version 0.6.4 very soon, as it seems very stable now.

I think, there is only one small thing left: merge #54 and update the documentation and notes to ensure everything is fine.

If there are any serious issues left, let's add them into the milestone #1 and resolve them until make a release?

@sezero, what do you think yourself?

After the 0.6.4 release, I want to start some bigger updates like reworking the VGM backend and other parts. That will be a big update, and before to start these works, it's reasonable to release the current version as a stable.

Previous stable version was out a while ago and no new versions had been released after that time.

## Plan for 0.6.4 release:
- [x] Merge #54 and give a note about the logic change of the multichannel support.
- [x] #58 Verify the old version of 0.6.3 to try reproduce the crash and compare with latest version to see, was that problem got been fixed or not.
- [x] ~~Before the release, add a note about the behaviour change after #58.~~ No longer needed, as the problem itself has been resolved.
- [x] Make the building of both static and shared builds to work.

**Comment by sezero** [comment]
> I think, it's a time to release the version 0.6.4 very soon, as it seems very stable now.

Yes

> I think, there is only one small thing left: merge #54 and update the documentation and notes to ensure everything is fine.

Yes

There is #58 but I don't know whether it is still an issue

**Comment by sezero** [comment]
Also: Is unrar support truly necessary in library side? IMO, such things should be handled in player programs, not library. (I won't fight for it though.)

**Comment by Wohlstand** [comment]
By default, unrar support is disabled. Do you think I should remove it from the library side? (do a small part of work doing in #74)

**Comment by sezero** [comment]
> By default, unrar support is disabled.

Unfortunately no, it is enabled by default

> Do you think I should remove it from the library side before 0.6.4 release?

Possibly yes: If nothing else, unrar is a 'non-free' code.

> (do a small part of work doing in #74)

Yes, unrar support can be further worked there, preferably in player side

**Comment by Wohlstand** [comment]
> Unfortunately no, it is enabled by default

Damn... I think, I missed something, ye, I see, it's enabled.

**Comment by Wohlstand** [comment]
> Yes, unrar support can be further worked there, preferably in player side

Ye, anyway, to be quick, I'll just set it disabled by default with adding a warning that this library is unfree.

**Comment by Wohlstand** [comment]
Done, I disabled it by default.

**Comment by myQwil** [comment]
#74 now removes unrar from the library end and re-implements it as an optional feature for the player.

**Comment by Wohlstand** [comment]
P.S. I forgot to append some: the 0.6.4 version I want to be a kind of a checkpoint release, because after it, I going to start some other works like improvements of the VGM support and other things.

**Comment by sezero** [comment]
>   0.6.4 version I want to be a kind of a checkpoint release,

Do you want #74 in 0.6.4 then? At least in minimal form which moves unrar support from library to player side?

**Comment by Wohlstand** [comment]
> Do you want https://github.com/libgme/game-music-emu/pull/74 in 0.6.4 then? At least in minimal form which moves unrar support from library to player side?

Would be good, yeah.

**Comment by sezero** [comment]
> > Do you want #74 in 0.6.4 then? At least in minimal form which moves unrar support from library to player side?
> 
> Would be good, yeah.

PING @myQwil then..

**Comment by myQwil** [comment]
It's basically done. The zip reader doesn't read .vgz files in .zip files as that's a 2nd layer of compression. You could always decompress the .vgz files individually and then repackage them in a new zip file.

**Comment by myQwil** [comment]
If you like, I can take out the zip reader for the time being.

**Comment by Wohlstand** [comment]
> If you like, I can take out the zip reader for the time being.

Ye, you could make a shot for that.

**Comment by sezero** [comment]
@Wohlstand: As far as I can see, #74 is ready for merging for 0.6.4

**Comment by Wohlstand** [comment]
**Some news:**
Sorry for the monthly delay, I was ill (possibly because of a new kind of flu that gets spread around: several of my buddies from different cities or even countries, also ill) and therefore, was being mostly unavailable. I going to resume any works very soon and finish the existing stuff.

**Comment by sezero** [comment]
> Sorry for the monthly delay, I was ill

Great that you recovered!

**Comment by Wohlstand** [comment]
About the #58, I actually found the reason of the problem, and I reproduced it locally on my Linux Mint, I had to build the latest VLC from the source code (the current version of VLC 3.0.16 at my Mint 21.3 doesn't triggers the problem, but the latest 3.0.20 built from the source does). I guess, after closing it, I can finally perform a release of 0.6.4 and start bigger works over the VGM thing.

**Comment by sezero** [comment]
> About the #58, I actually found the reason of the problem, and I reproduced it locally on my Linux Mint,

That one should be considered fixed, so I guess you can tick the tickbox above

**Comment by sezero** [comment]
FYI: I have been maintaining a branch forked from v0.6.3 release and applying only bug fixes and build clean-ups: https://github.com/libsdl-org/game-music-emu/tree/v0.6.3-SDL or https://github.com/sezero/game-music-emu/tree/0.6.3-fixes

It is very stable and we have been using that one in SDL_mixer in _vendored_ builds

**Comment by sezero** [comment]
Before doing a 0.6.4 release, I guess the mess from https://github.com/libgme/game-music-emu/pull/92 should be fixed, or it should be reverted.

**Comment by Wohlstand** [comment]
> I guess the mess from #92 should be fixed

Could you summarize what's a problem of that right now if something also than the method of building shared and static things.

**Comment by sezero** [comment]
> > I guess the mess from #92 should be fixed
> 
> Could you summarize what's a problem of that right now if something also than the method of building shared and static things?

Not being able to build both static and shared libs. Can't remember if there were any other issues.

**Comment by Wohlstand** [comment]
Okay then, I going to perform some today to resolve that, and result I'll put on the second branch and if everything is fine, then to mainstream.

**Comment by Wohlstand** [comment]
Hello!
I will release 0.6.4 once I will resolve the static+shared build problem, I'll try to perform this close to days off, as now I am busy on the main work. I am always available online by many ways, so, feel free to ask any questions, I will reply as soon as I can.

**Comment by Wohlstand** [comment]
Because our time zones are close :)

**Comment by Wohlstand** [comment]
@sezero, I made just now the fix of the last item https://github.com/libgme/game-music-emu/pull/115, and if no other SERIOUS problems, I will release 0.6.4 after that PR will be merged.

**Comment by Wohlstand** [comment]
Mission complete! https://github.com/libgme/game-music-emu/releases/tag/0.6.4
Now, going to work for new features very soon!

**Comment by Wohlstand** [comment]
(I might release minor updates if any critical issues would be found at any moment)
